### DISCUSSION TITLE
Make the app UI a bit modern like a normal Android app, and adapt Material 3 design.

### DISCUSSION THREAD

**Discussion by Mesum-Hussain** [termux/termux-app#3199] [Ideas]
I get it that it's a terminal emulator for Android made in an attachment to bring GNU Linux functionality to Android as much as possible, but I don't think it needs to look exactly like a desktop terminal emulator with a very ugly traditional UI.

It can look like an instant messaging app where we type the command in a message text field, and when we press the "send" arrow button, it executes the command. That way we can also allow users to use swipe typing (could be kept optional if users don't want it).

 And the output can appear in the form of the UI of a Jupyter Notebook.

And then everything can be made to follow the Theming of Material 3 design with dynamic colors, etc. 

I also think that the **"New Session"** doesn't need a navigation drawer dedicated to it. It can be handled as Tabs are handled in the Google Chrome Browser app. This will make the UI more minimalistic and clean. 

Somewhat like this:

![IMG_20230121_140456.png](https://user-images.githubusercontent.com/64641670/213875659-8ebee758-e0d3-485d-9af7-72d0a0780985.png)

Please pardon the bad proportions..... made this mock up on my phone on the go.

**Comment by agnostic-apollo** [comment]
>where we type the command in a message text field, and when we press the "send" arrow button, it executes the command. That way we can also allow users to use swipe typing (could be kept optional if users don't want it).

Already supported.

https://wiki.termux.com/wiki/Touch_Keyboard#Text_Input_View

>doesn't need a navigation drawer dedicated to it. It can be handled as Tabs are handled in the Google Chrome Browser app. This will make the UI more minimalistic and clean.

Chrome has a tabs menu too, same thing as a drawer. Moreover, terminal requires full screen to take as much space as possible, same as when you scroll in chrome. Moreover, you can easily use shortcuts to create and move between sessions.

https://wiki.termux.com/wiki/Terminal_Settings#Handling_terminal_sessions

**Comment by Mesum-Hussain** [comment]
> >where we type the command in a message text field, and when we press the "send" arrow button, it executes the command. That way we can also allow users to use swipe typing (could be kept optional if users don't want it).
>
> Already supported.
> 
> https://wiki.termux.com/wiki/Touch_Keyboard#Text_Input_View

I am aware..... talking about the UI over here. And by the way it's not very intuitive to find in the first place..... I got to know about it very recently when I swiped the 'extra keys ribbon' to the left *by mistake*. 

> >doesn't need a navigation drawer dedicated to it. It can be handled as Tabs are handled in the Google Chrome Browser app. This will make the UI more minimalistic and clean.
> 
> Chrome has a tabs menu too, same thing as a drawer.

Again talking about improving the UI over here..... though I agree that Chrome has the same functionality too, I think Chrome's implementation is better in this case considering the Phone form factor. We don't need a complete navigation drawer just to add a new tab. 

>Moreover, terminal requires full screen to take as much space as possible, same as when you scroll in chrome.

I was thinking of a design somewhat like a Jupyter Notebook..... where the command is in a small box and it's output is presented below it, and both of them are enclosed in a big box indicating a single command interaction between the user and the shell. New big box for the next command. I don't know about its feasibility.....do we really need to display a real terminal in its default look? Can't we take its results and present it in a more modern looking frontend interface? I wonder how Jupyter Notebook is tackling this problem with the Python Shell.

>Moreover, you can easily use shortcuts to create and move between sessions.
> 
> https://wiki.termux.com/wiki/Terminal_Settings#Handling_terminal_session

Again I am not complaining about the lack of functionality.....my issue is with the UI. It really doesn't need to be that ugly. We can design it to look like a normal Android app with all the functionality of the terminal.

**Reply by RalfWerner** [reply]
I am glad that the Termux looks like a **normal** _terminal_, uses almost the complete small mobile screen and not like an Android app. If you don't like this, you can use the _plugins_ Termux:API, Termux:GUI, Termux:Widget or Termux:Float. With them you can probably fulfill all of your Android wishes. To do this, I recommend using **normal** Termux

**Reply by Mesum-Hussain** [reply]
> I am glad that the Termux looks like a **normal** _terminal_, uses almost the complete small mobile screen and not like an Android app. If you don't like this, you can use the _plugins_ Termux:API, Termux:GUI, Termux:Widget or Termux:Float. With them you can probably fulfill all of your Android wishes. To do this, I recommend using **normal** Termux 

Good for you.

I personally prefer the Jupyter Notebook style Shell Interface, it looks more modern in my opinion.

I personally don't see why we need to have an old school, 30 year old terminal-like Shell Interface when the modern computers are **way more than capable** of handling Graphical User Interfaces..... I really don't see any utility to it other than just personal preference (in fact it's rather scaring off beginners in my opinion..... it's contributing to a bad UX).

Anyways thanks for suggesting, I will check those out.

**Reply by agnostic-apollo** [reply]
Are you one of those window's guy who has never heard of the cmd?

> And by the way it's not very intuitive to find in the first place..... 

If you read the wiki, you should have found it. There is need for improving the docs though, which is being worked on, but users still must read the manual.

https://wiki.termux.com/wiki/User_Interface

>We don't need a complete navigation drawer just to add a new tab.

"You" may not need it, others do, some people don't even want the status bar showing and use fullscreen mode. Support could be added in future for a bar at top, but not a priority since can just use shortcuts, probably extra keys custom macro may work too.

>do we really need to display a real terminal in its default look? 

Termux is a *"terminal app"*, for it to not have a terminal (by default) will make it into a *"not a terminal app"*. You may want to find or create other apps that provide other GUI interfaces while using termux binaries with termux apis like RUN_COMMAND.

>where the command is in a small box and it's output is presented below it, and both of them are enclosed in a big box indicating a single command interaction between the user and the shell.

That is certainly possible by using termux's background [`AppShell`](https://github.com/termux/termux-app/blob/2f5a6f7de663074df829a62e44a23ee70e932763/termux-shared/src/main/java/com/termux/shared/shell/command/runner/app/AppShell.java), I even have it in one of one personal test apps. But `AppShell` only returns results when commands are complete, so can't view current progress. It needs to be rewritten in future for such support. Moreover, adding such support in Termux main activity to add other type of tabs instead of a terminal would require a major refactor and lot of time. And since its not a high priority feature or even requested by lot of users, unlikely to be done anytime soon, if ever.

> It really doesn't need to be that ugly.

It doesn't look ugly to me and you can change colors and fonts too. There are also pull submitted for other improvements.

>We can design it to look like a normal Android app with all the functionality of the terminal.

Again, termux is a "terminal app", we don't want it to be like just a normal app.

>I personally don't see why we need to have an old school, 30 year old terminal-like Shell Interface when the modern computers are way more than capable of handling Graphical User Interfaces..... I really don't see any utility to it other than just personal preference

You stance is because you don't understand the complexity of a terminal and how powerful it is. Why do you think terminals have been worked on for decades if only 2 text boxes were enough for running shell commands, specially 30 years ago when computers were not powerful at all? By did people bother? You should research into the features provided by a terminal. Its a gazillion times faster for many things than a GUI, and a gazillion times slower at other things.

> (in fact it's rather scaring off beginners in my opinion..... it's contributing to a bad UX).

Programming is hard and to be a good programmer, one must understand how code or shell commands runs underneath, and a terminal is a very good way to understand that. Good universities use terminals too. Abstracting away such stuff for each is harmful in the long run.

Your way of running one command and showing its output can also easily be done in the terminal with a script/c code and the `clear` command anyways.

**Reply by Mesum-Hussain** [reply]
> Are you one of those window's guy who has never heard of the cmd?

Insults are not help. Sorry. 
CMD has a bad UI, Jupyter Notebook UI is a lot better for a Shell.

> > And by the way it's not very intuitive to find in the first place..... 
> 
> If you read the wiki, you should have found it. There is need for improving the docs though, which is being worked on, but users still must read the manual.
> 
> https://wiki.termux.com/wiki/User_Interface

I didn't read or even know it existed in the first place to be honest. Not everybody reads manuals, in fact that's the normal behaviour, products should be made simple for all kinds of users, needing to read manuals to figure out basic stuff is **very bad** UX..... but that's not the point of the discussion though.

> >We don't need a complete navigation drawer just to add a new tab.
>
> "You" may not need it, others do, some people don't even want the status bar showing and use fullscreen mode. Support could be added in future for a bar at top, but not a priority since can just use shortcuts, probably extra keys custom macro may work too.

I hope you realise "we" was metaphorical..... what I meant was that Chrome's implementation of Tabs is better for a mobile form factor. You may disagree with me though, no issues, it's just my opinion.

> >do we really need to display a real terminal in its default look? 
> 
> Termux is a *"terminal app"*, for it to not have a terminal (by default) will make it into a *"not a terminal app"*.

Termux is useful for its functionality, not for it's "Terminal-like" UI. Makes me wonder why do we want to stay so true to the Terminal-like UI? I mean this is a new kind of device..... Not a 30 year old traditional desktop, why aren't we adapting the Terminal to suit this kind of device? Why are we forcing a desktop app to squeeze into a mobile form factor with a subpar typing experience? Honestly without swipe typing or voice typing, smartphone keyboards are atrocious!

> >where the command is in a small box and it's output is presented below it, and both of them are enclosed in a big box indicating a single command interaction between the user and the shell.
> 
> That is certainly possible by using termux's background [`AppShell`](https://github.com/termux/termux-app/blob/2f5a6f7de663074df829a62e44a23ee70e932763/termux-shared/src/main/java/com/termux/shared/shell/command/runner/app/AppShell.java), I even have it in one of one personal test apps. But `AppShell` only returns results when commands are complete, so can't view current progress. It needs to be rewritten in future for such support. Moreover, adding such support in Termux main activity to add other type of tabs instead of a terminal would require a major refactor and lot of time. And since its not a high priority feature or even requested by lot of users, unlikely to be done anytime soon, if ever.

Hope it happens though. But no issues if it doesn't. Just thought it might make an interesting discussion so posted here. And who knows, it might improve Termux to become a next gen Shell app, so just posted.

> > It really doesn't need to be that ugly.
> 
> It doesn't look ugly to me and you can change colors and fonts too. There are also pull submitted for other improvements.

What I mean is that morden Android apps are so much better looking with all that Material You Theming and design guidelines.....a Terminal is the most basic design from 30 years ago which was designed keeping in mind the computing limitations of that era..... Modern computers have none of those limitations for us to keep that UI around. It's just a convention at this point, which was still fine on a desktop, but becomes very annoying on a smartphone.

> >We can design it to look like a normal Android app with all the functionality of the terminal.
> 
> Again, termux is a "terminal app", we don't want it to be like just a normal app.

It's useful for its functionality, not for its UI. We don't need to stay true to the "Terminal". We should improve old useful things to adapt to the modern world. Think from the First Principals..... What's the utility of being a "Terminal Interface" on a touchscreen device of a phone form factor? Honestly I personally think staying true to the "Terminal" is holding Termux back from becoming an awesome Android Shell app. 

> >I personally don't see why we need to have an old school, 30 year old terminal-like Shell Interface when the modern computers are way more than capable of handling Graphical User Interfaces..... I really don't see any utility to it other than just personal preference
> 
> You stance is because you don't understand the complexity of a terminal and how powerful it is. Why do you think terminals have been worked on for decades if only 2 text boxes were enough for running shell commands, specially 30 years ago when computers were not powerful at all?

All I know is that Jupyter Notebook has the same UI and it's literally capable of doing any kind of coding using the Python Shell. Don't see why Termux can't have such a UI and still remain as useful as it ever was.

>You should research into the features provided by a terminal. Its a gazillion times faster for many things than a GUI, and a gazillion times slower at other things.

Not when you have to type out every single character by two thumbs on a virtual keyboard on a touchscreen device. Typing 2 minutes to get a 2 second speed boost..... Doesn't seem ideal to me. Swipe typing or voice typing could ***drastically*** improve the UX.

> > (in fact it's rather scaring off beginners in my opinion..... it's contributing to a bad UX).
> 
> Programming is hard and to be a good programmer, one must understand how code or shell commands runs underneath, and a terminal is a very good way to understand that. Good universities use terminals too. Abstracting away such stuff for each is harmful in the long run.

Well you see, Python is more popular than C because it's easy, even though C is so much better as a language. I understand your philosophy, but improving accessibility when the technology can handle it, is the right move forward in my opinion. 

People will still keep learning about the terminal, we are not making it obsolete.....we are just improving the experience of interacting with the Bash  Shell on Android in the Termux app, because the Terminal UI is not ideal for the mobile form factor.

> Your way of running one command and showing its output can also easily be done in the terminal with a script/c code and the `clear` command anyways.

I am not asking to clear the previous commands..... just like Jupyter Notebook basically.....new Cell for next command..... somewhat like sending a message on an instant messaging app..... like chatting with the Shell.

**Comment by RalfWerner** [comment]
@Mesum-Hussain You seem to be a fan of Jupyter Notebook. This can be installed and used on your Android device with the help of Termux. I found a [YT video](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjG4L6tmN38AhUGuaQKHWYVDPoQtwJ6BAggEAI&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DvfdBTYiwyWs&usg=AOvVaw3KqNxWlwyArXW3uSYatl_0) to install and run (7:14) it. However, the author had no problems with the appearance of Termux. Maybe that's your problem solution

**Reply by Mesum-Hussain** [reply]
> @Mesum-Hussain You seem to be a fan of Jupyter Notebook. This can be installed and used on your Android device with the help of Termux. I found a [YT video](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjG4L6tmN38AhUGuaQKHWYVDPoQtwJ6BAggEAI&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DvfdBTYiwyWs&usg=AOvVaw3KqNxWlwyArXW3uSYatl_0) to install and run (7:14) it. However, the author had no problems with the appearance of Termux. Maybe that's your problem solution

It's not a problem as such, I just think adopting the Jupyter Notebook UI is more suitable for an Android Sell app..... that would make it a perfect Android Shell app. The current UI is really not optimal if you think from the first principles.....it was designed for ***very weak*** desktop computers about 40 years ago, but modern mobiles are neither weak nor desktop computers..... it's a completely different product and its app should be made keeping that in mind.

I posted it here because I thought it was an interesting topic for discussion. No problem if nothing comes out of this discussion. But if in case it influences Termux somehow, I would be very delighted.

The only technical problem I currently have on Termux is that swipe typing or voice typing don't work..... which is not ideal for typing on a smartphone using a virtual keyboard.

**Comment by luisdavim** [comment]
This would limit the usability a lot, specially if you consider the Dex and other desktop modes. And tablets.  My main use case for this app is to use my large screen tablet to write code using vim.

**Reply by Mesum-Hussain** [reply]
> This would limit the usability a lot, specially if you consider the Dex and other desktop modes. And tablets.  My main use case for this app is to use my large screen tablet to write code using vim.

Although it's completely your presence and right to use what you like for your work, I personally think [Acode](https://github.com/deadlyjack/Acode) is a better tool for this job..... I write the code in Acode and then execute it using Termux, just like we do on desktop using VS Code and the Terminal. 

But even in that case I don't think anything will change. 

Currently we see a Shell Prompt to which we give commands which the Shell executes and gives an output..... when we use Vim, we command the Shell to run Vim in response to which it clears the entire screen and gives control to Vim. We write something, save and exit, to get back to the same old Shell Prompt to run our next command.

Same can be done here too.....we can have this as the Shell Interface which runs commands, and when we run something like Vim, the Shell clears the entire UI and give control to Vim.

I don't know how Termux is handling rendering, but I am pretty sure this can be done with something like Flutter (so at least it's a realistically possible concept).

By the way I don't think that this UI is only optimal for Candy-Bar Phones, I think this UI will work well even in case of large-screen devices. I mean Jupyter Notebook works well on desktop screen too, why would anything be different in case of Termux?

I just think this is a better, more modern looking Shell interface, and all the Shell applications on all kinds of devices should abandon the "Terminal" interface if favour of a new UI something like this. "Terminal" interface is just a convention with which people are too accustomed to that they don't want to change it..... otherwise I really don't see any utility of it in today's world where Computing Power and Storage is not a significant issue. 
I mean [TTY interface](https://en.m.wikipedia.org/wiki/Teleprinter) was literally designed and used for the computers which didn't have screens back in the day (now we have a considerably big Touchscreen as an entire computer AKA Tablet), we are still following that convention almost blindly.

**Comment by mz71** [comment]
I'm using a physical keyboard and a terminal multiplexer. How would it work in your design? I want as much working space as possible, because the phone screen is so small. 
![Screenshot_20230204-005734_Termux](https://user-images.githubusercontent.com/20141904/216732722-34a8f5ca-8ee3-41fe-9164-e536478cd90c.jpg)

There are so. many. tools. geared specifically towards CLI. And lately there's even more popping up. I feel the UX of the terminal is basically a finalized thing - your idea is to replace it with something else, and that just constitutes a new app imo. You just can't change the terminal interface without breaking changes.

And lastly, some people just love CLI more than any GUI. They don't like point&click and Termux is a perfect app for them. Don't fall into the trap that material design will be perfect for everyone and everything on Android.

**Reply by Mesum-Hussain** [reply]
> I'm using a physical keyboard and a terminal multiplexer. How would it work in your design? I want as much working space as possible, because the phone screen is so small. 
> ![Screenshot_20230204-005734_Termux](https://user-images.githubusercontent.com/20141904/216732722-34a8f5ca-8ee3-41fe-9164-e536478cd90c.jpg)
> 

If you are confused about my proposed design, then let me clear it for you: It's a **Shell Interface** design. It's not changing anything, it's just changing the look of the UI. People have been using different Shell Prompts successfully for a long time:

![44c8b83efefd16f8e81efc236ff74321.png](https://user-images.githubusercontent.com/64641670/216761921-f6f9a209-3310-4e2c-9af4-4bede8b58991.png)

![b458f62e5eb5f9e631e4ea160b0b466e.jpeg](https://user-images.githubusercontent.com/64641670/216761986-58f35513-68c9-448e-b87e-0a0123a9c19f.jpeg)

My design does nothing different than these except for using the graphical capabilities of the morden computers to render a more morden GUI. 

If you use a multiplexer, then according to my design, you will see multiple windows of the Shell interface (which looks like Jupyter Notebook) to run your commands. Just like you see multiple windows having Shell prompts in the current UI. In theory, it's going to break nothing..... Only the UI will improve according to modern standards.



> There are so. many. tools. geared specifically towards CLI. And lately there's even more popping up. I feel the UX of the terminal is basically a finalized thing - your idea is to replace it with something else, and that just constitutes a new app imo. You just can't change the terminal interface without breaking changes.

It's not a new thing which is changing the fundamentals..... It's just a design suggestion which is trying to improve the Shell interface according to modern standards using the modern capabilities of the computers.

> And lastly, some people just love CLI more than any GUI. They don't like point&click and Termux is a perfect app for them.

No it's not. Termux is an Android app, and Android is a Touch-First OS..... If they prefer typing more than clicking (touching), then they are on the wrong OS. Their experience will be much better on something like Linux with a Desktop Terminal app. Similarly, Android user's experience will be much better with a Shell app custom designed for the touchscreen devices than using a desktop Shell app forcefully squeezed into an APK.

**Reply by Mesum-Hussain** [reply]
Besides, it's just an idea at this point.....we can discuss and adapt it to different use cases: using with a multiplexer, using it on a big screen, etc..... We can come up with a design which is suitable for all use cases and still remain appropriate according to the modern design and UX standards.

Termux is a new kind of product (a Shell app for Touchscreen devices), which will inevitably require changes and innovation to make it the best Shell app for Touchscreen. We cannot avoid that if we want to provide a good UX. 

And as we know iPad is shipping with an M2 chip and Stage Manager, and Google is Adapting Android to the large screen devices, and ARM chips are catching up with x86, it's kinda obvious that a major part of the computing devices of the near future would consist of tablets..... The needs of the majority of the people who needed laptop earlier, would be satisfied by a tablet soon.

Innovation in Termux UI is probably going to determine the future of coding on the computers of the future (Tablets).

**Reply by mz71** [reply]
I don't think you can just put a gui textbox instead of a prompt. In the end, it still would need to hinge on a text basesd terminal, so you would need to alter the terminal output from the user (e.g. move the prompt into the gui textbox). That would make many cli software incompatible (like the terminal multiplexer zellij that i'm using) because they expect the prompt inside the terminal.

The terminal operates on text, the whole working area is a text canvas. The shell prompt is thus rendered as text.

> Termux is an Android app, and Android is a Touch-First OS..... If they prefer typing more than clicking (touching), then they are on the wrong OS. Their experience will be much better on something like Linux with a Desktop Terminal app

While I agree that we "may be on the wrong OS"... some things to consider
1. In many parts of the world, an Android phone and a physical keyboars is easier to buy than a Linux laptop. Termux empowers many people to learn Linux and CLI on Android in their raw form.
2. Not every program on a given OS needs to have a consistent feel to it. Termux UX is not like other apps,but this is a deliberate choice. Cold terminal down to the core - that's the design principle of Termux. It's refreshing to see an Android app that doen't comply to its UI model, for me it's a sign of a healthy OS with many worthwhile apps.
3. Your project is perfect for a new app. I use Helix, a CLI tool for programming, while you use Acode. If you tried to ask on Helix github "Make the app UI a bit modern" they would not even think about it - because they want a CLI app. And you want a GUI app. That's why you use Acode and not Vim for example. And that's fine, it is great that we have two different apps for this. For terminals, we don't have this luxury - but in the future, "Ashell" could thrive!

**Reply by Mesum-Hussain** [reply]
> I don't think you can just put a gui textbox instead of a prompt. In the end, it still would need to hinge on a text basesd terminal, so you would need to alter the terminal output from the user (e.g. move the prompt into the gui textbox). That would make many cli software incompatible (like the terminal multiplexer zellij that i'm using) because they expect the prompt inside the terminal.

Hmm .... that's a valid issue to handle I guess. Let's see how we can tackle this: I am dividing all the terminal software into 3 categories..... CLI, TUI, and something like a Multiplexer..... Handling CLI apps is a non issue because Jupyter Notebook Design does it perfectly right now..... Handling TUI apps (like Vim, Htop, etc. ) could be done by giving the entire screen to the Terminal software when it is running (just like how it's done right now)..... Remaining is Multiplexer, which I guess could be integrated into the Termux app Settings itself as an Option or perhaps as an Extension (like there are other Extensions for changing theme and floating window).

> The terminal operates on text, the whole working area is a text canvas. The shell prompt is thus rendered as text.

And that's what we are trying to change, because we are no longer limited to a computer which is only capable of handling Text, in the present time. I am sure we can find an innovative solution even to this problem.

> > Termux is an Android app, and Android is a Touch-First OS..... If they prefer typing more than clicking (touching), then they are on the wrong OS. Their experience will be much better on something like Linux with a Desktop Terminal app
> 
> While I agree that we "may be on the wrong OS"... some things to consider
> 1. In many parts of the world, an Android phone and a physical keyboars is easier to buy than a Linux laptop. Termux empowers many people to learn Linux and CLI on Android in their raw form.

Imagine not even needing a physical keyboard in that situation. That's what I want to achieve with this proposal..... I want Termux to be so tailored for Android that just the phone should be enough to get the best Shell experience. Everything should be done with touches and gestures. That would be the true empowerment of an Android user using Termux in a remote area in a third world country.

>Cold terminal down to the core - that's the design principle of Termux. It's refreshing to see an Android app that doen't comply to its UI model, for me it's a sign of a healthy OS with many worthwhile apps.

From my perspective Termux is an Android Shell app, and to the best Android Shell app, it requires to provide a better experience with gesture compared to with a keyboard attachment. If we require an attachment to get a better experience, then that app is not a good native app. Might as well run Linux apps in a VM and pretend that it's an Android app, but will it actually be an Android app though? 
I don't know about Termux team, but for me if an Android app is better with a keyboard and mouse (as opposed to being better with just a touchscreen, without a keyboard and mouse) then it's not a good Android app..... it's rather a good Desktop app. 

Although it's refreshing to see variety of apps on Android, I would actually prefer if they are real Android apps (optimised for touchscreen) as opposed to being a desktop app ported as an APK. Following the native design language is a bonus in my perspective though.

> 3. Your project is perfect for a new app. I use Helix, a CLI tool for programming, while you use Acode. If you tried to ask on Helix github "Make the app UI a bit modern" they would not even think about it - because they want a CLI app. And you want a GUI app. That's why you use Acode and not Vim for example. And that's fine, it is great that we have two different apps for this. For terminals, we don't have this luxury - but in the future, "Ashell" could thrive!

Maybe you are right. I actually wasn't really expecting that this idea would come through, but I thought it's an interesting idea so it's worthwhile to post anyway. Maybe by some chance it might actually influence Termux. 

Though I would love to see such an app come into existence..... would be awesome opening my Foldable Tablet to code a project in a Metro (swipe typing is even better with a stylus). I would have created it myself if I had skills.

**Comment by Alephaxe** [comment]
i like the original design, there is no need in overloaded material gui
